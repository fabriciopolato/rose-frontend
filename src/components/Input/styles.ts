import styled from 'styled-components';

export const StyledInput = styled.input`
  background-color: ${props => props.theme.white};
  border: 0.8px solid rgba(0, 0, 0, 0.1);
  border-radius: 8px;
  padding: 8px 0 8px 16px;

  font-family: Raleway;
  font-weight: normal;
  font-size: 1.6rem;

  color: ${props => props.theme.black};

  &::placeholder {
    color: #333;
  }
`;
