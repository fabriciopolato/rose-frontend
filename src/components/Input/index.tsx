import React, { InputHTMLAttributes } from 'react';

import { StyledInput } from './styles';

const Input: React.FC<InputHTMLAttributes<HTMLInputElement>> = ({
  children,
  ...rest
}) => {
  return <StyledInput {...rest} />;
};

export default Input;
