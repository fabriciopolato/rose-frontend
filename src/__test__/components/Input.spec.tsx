import React from 'react';
import { render } from '@testing-library/react';

import Input from '../../components/Input';

describe('<Input />', () => {
  it('should match previous snapshot', () => {
    const { asFragment } = render(<Input />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have basic input attributes provided by props', () => {
    const { getByPlaceholderText } = render(
      <Input type="text" placeholder="Placeholder Test" />
    );

    const inputElement = getByPlaceholderText(
      'Placeholder Test'
    ) as HTMLInputElement;

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute('type', 'text');
  });
});
