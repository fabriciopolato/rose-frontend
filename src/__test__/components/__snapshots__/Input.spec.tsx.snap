// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match previous snapshot 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm bHsGLn"
  />
</DocumentFragment>
`;
